server/game/core/ongoingEffect/effectImpl/DynamicOngoingEffectImpl.ts

```diff
@@ -35,20 +67,18 @@ export default class DynamicOngoingEffectImpl<TValue> extends StaticOngoingEffec
             (Array.isArray(oldValue) && Array.isArray(newValue)) ||
             (typeof oldValue === 'object' && typeof newValue === 'object')
         ) {
-            return JSON.stringify(oldValue) !== JSON.stringify(newValue);
+            const replacer = (key: string, value: any) => {
+                if (value instanceof GameObject) {
+                    return value.uuid;
+                }
+                return value;
+            };
```

Comment:
This avoids circular references because all `GameObject` have a reference to `game` and `game` has a reference to all `GameObject`:
```
    TypeError: Converting circular structure to JSON
        --> starting at object with constructor 'Game'
        |     property 'ongoingEffectEngine' -> object with constructor 'OngoingEffectEngine'
        --- property 'game' closes the circle
```